TestAssets/TestProjects/DesktopMinusRid/Program.cs

```diff
@@ -2,14 +2,41 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.InteropServices;
 
 namespace TestApp
 {
-    public class Program
+    class Program
     {
-        public static void Main(string[] args)
+        static void Main()
         {
+            // Prevent libuv on path from interfering with test
+            Environment.SetEnvironmentVariable("PATH", "");
+
+#if USE_NATIVE_CODE
+            try
+            {
+                uv_loop_size();
+                Console.WriteLine($"Native code was used ({GetProcessorArchitecture()})");
+            }
+            catch (DllNotFoundException)
+            {
+                Console.WriteLine($"Native code failed ({GetProcessorArchitecture()})");
+            }
+#else      
+            Console.WriteLine($"Native code was not used ({GetProcessorArchitecture()})");
+#endif  
+        }
+
+#if USE_NATIVE_CODE
+        [DllImport("libuv", CallingConvention = CallingConvention.Cdecl)]
+        static extern int uv_loop_size();
+#endif
+
+        static ProcessorArchitecture GetProcessorArchitecture()
+        {
+            return AssemblyName.GetAssemblyName(Assembly.GetExecutingAssembly().Location).ProcessorArchitecture;
```

Comment:
I'd suggest `typeof(Program).Assembly.Location`.  I don't know that it matters at all, but `Assembly.GetExecutingAssembly()` is one of the methods that we originally removed from .NET Core because it doesn't play well with optimizations we might want to do (inlining across assembly boundaries).